tinygrad/shape/shapetracker.py

```diff
@@ -18,7 +18,9 @@ def un1d(shape:Tuple[sint, ...], offs:sint) -> List[sint]:
 
 @functools.lru_cache(maxsize=None)
 def merge_views(vm2:View, vm1:View) -> Optional[View]:
-  if vm1.contiguous and vm1.shape == vm2.shape: return vm2
+  if vm1.contiguous:
+    if vm1.shape == vm2.shape: return vm2
+    if vm1.size() == vm2.size() and (vm2_new := vm2.reshape(vm1.shape)) is not None: return vm2_new
```

Comment:
Also do you have to check the size here? Won't the reshape always fail if they are different sizes?

Reply:
Thats correct. But isn't reshape costlier (it would call merge_dims and that woud return None when it can't merge the dims...), so was trying to avoid calling it, thats why the size equality check.

Reply:
Speed is never worth complexity!

Reply:
tried removing it, realised reshape throws ValueError when the sizes don't match!